mmcv/runner/hooks/logger/base.py

```diff
@@ -37,23 +43,46 @@ def before_epoch(self, runner):
     def after_train_iter(self, runner):
         if self.every_n_inner_iters(runner, self.interval):
             runner.log_buffer.average(self.interval)
+            self.sync_buffer_output(runner)
         elif self.end_of_epoch(runner) and not self.ignore_last:
-            # not precise but more stable
             runner.log_buffer.average(self.interval)
+            self.sync_buffer_output(runner)
+        if runner.log_buffer.ready:
+            self.log(runner)
+            if self.reset_flag:
+                runner.log_buffer.clear_output()
 
+    def after_val_iter(self, runner):
+        if self.every_n_inner_iters(runner, self.interval):
+            runner.log_buffer.average(self.interval)
+            self.sync_buffer_output(runner)
+        elif self.end_of_epoch(runner) and not self.ignore_last:
+            runner.log_buffer.average(self.interval)
+            self.sync_buffer_output(runner)
         if runner.log_buffer.ready:
             self.log(runner)
             if self.reset_flag:
                 runner.log_buffer.clear_output()
 
     def after_train_epoch(self, runner):
+        runner.log_buffer.average()
+        self.sync_buffer_output(runner)
         if runner.log_buffer.ready:
             self.log(runner)
             if self.reset_flag:
                 runner.log_buffer.clear_output()
 
     def after_val_epoch(self, runner):
         runner.log_buffer.average()
-        self.log(runner)
-        if self.reset_flag:
-            runner.log_buffer.clear_output()
+        self.sync_buffer_output(runner)
+        if runner.log_buffer.ready:
```

Comment:
Maybe there is no need to check `runner.log_buffer.ready` after `runner.log_buffer.average()`?